code/modules/paperwork/photography.dm

```diff
@@ -175,119 +175,141 @@
 	res.Scale(size*32, size*32)
 	// Initialize the photograph to black.
 	res.Blend("#000", ICON_OVERLAY)
+	CHECK_TICK
 
-	var/atoms[] = list()
-	for(var/turf/the_turf in turfs)
+	var/list/atoms = list()
+	for(var/turf/the_turf as anything in turfs)
 		// Add outselves to the list of stuff to draw
 		atoms.Add(the_turf);
 		// As well as anything that isn't invisible.
-		for(var/atom/A in the_turf)
-			if(A.invisibility) continue
-			atoms.Add(A)
+		for(var/atom/cur_atom as anything in the_turf)
+			if(!cur_atom || cur_atom.invisibility)
+				continue
+			atoms.Add(cur_atom)
 
 	// Sort the atoms into their layers
 	var/list/sorted = sort_atoms_by_layer(atoms)
 	var/center_offset = (size-1)/2 * 32 + 1
-	for(var/i; i <= sorted.len; i++)
-		var/atom/A = sorted[i]
-		if(A)
-			var/icon/IM = getFlatIcon(A)//build_composite_icon(A)
-
-			// If what we got back is actually a picture, draw it.
-			if(istype(IM, /icon))
-				// Check if we're looking at a mob that's lying down
-				if(istype(A, /mob/living))
-					var/mob/living/L = A
-					if(!istype(L, /mob/living/carbon/xenomorph)) //xenos don't use icon rotatin for lying.
-						if(L.lying)
-							// If they are, apply that effect to their picture.
-							IM.BecomeLying()
-				// Calculate where we are relative to the center of the photo
-				var/xoff = (A.x - center.x) * 32 + center_offset
-				var/yoff = (A.y - center.y) * 32 + center_offset
-				if (istype(A,/atom/movable))
-					xoff+=A:step_x
-					yoff+=A:step_y
-				res.Blend(IM, blendMode2iconMode(A.blend_mode),  A.pixel_x + xoff, A.pixel_y + yoff)
+	for(var/atom/cur_atom as anything in sorted)
+		if(!cur_atom)
```

Comment:
use `if(!QDELETED())` here because the atoms might have been deleted during the various CHECK_TICK below

ideally you also want to check back that they are still in the picture frame aswell